lib/kadai-core/src/main/java/io/kadai/task/internal/TaskDistributor.java

```diff
@@ -0,0 +1,277 @@
+  }
+
+  public BulkOperationResults<String, KadaiException> distribute(
+      String sourceWorkbasketId, List<String> taskIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, taskIds, null, null, null);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithStrategy(
+      String sourceWorkbasketId,
+      List<String> taskIds,
+      String distributionStrategyName,
+      Map<String, Object> additionalInformation)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(
+        sourceWorkbasketId, taskIds, distributionStrategyName, null, additionalInformation);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithStrategy(
+      String sourceWorkbasketId,
+      String distributionStrategyName,
+      Map<String, Object> additionalInformation)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(
+        sourceWorkbasketId, null, distributionStrategyName, null, additionalInformation);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithDestinations(
+      String sourceWorkbasketId, List<String> destinationWorkbasketIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, null, null, destinationWorkbasketIds, null);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithDestinations(
+      String sourceWorkbasketId, List<String> taskIds, List<String> destinationWorkbasketIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, taskIds, null, destinationWorkbasketIds, null);
+  }
+
+  private BulkOperationResults<String, KadaiException> distributeTasks(
+      String sourceWorkbasketId,
+      List<String> taskIds,
+      String distributionStrategyName,
+      List<String> destinationWorkbasketIds,
+      Map<String, Object> additionalInformation)
+      throws InvalidArgumentException,
+          NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    BulkOperationResults<String, KadaiException> operationResults = new BulkOperationResults<>();
+
+    try {
+      kadaiEngine.openConnection();
+
+      workbasketService.checkAuthorization(sourceWorkbasketId, WorkbasketPermission.DISTRIBUTE);
+
+      if (taskIds == null) {
+        taskIds = getTaskIdsFromWorkbasket(sourceWorkbasketId);
+      }
+
+      if (taskIds.isEmpty()) {
+        return operationResults;
+      } else {
+        checkIfTasksInSameWorkbasket(taskIds);
```

Comment:
these two checks will also be performed, if no taskIds are provided and retrieved via the sourceWorkbasket, but we don't need it in this case, right?